scripts/__input_class_gamepad/__input_class_gamepad.gml

```diff
@@ -236,13 +235,25 @@ function __input_class_gamepad(_index) constructor
         {
             if (__haptic_curve != undefined)
             {
-                var _struct = animcurve_get(__haptic_curve);
-            
-                __haptic_strength_left  = animcurve_channel_evaluate(_struct.channels[0], __haptic_step/__haptic_steps_total);
-                __haptic_strength_right = animcurve_channel_evaluate(_struct.channels[((array_length(_struct.channels) > 1)? 1 : 0)], __haptic_step/__haptic_steps_total);
+                var _animation_curve = animcurve_get(__haptic_curve);
+                var _channel_count   = array_length(_animation_curve.channels);
+                
+                if (_channel_count == 1)
+                {
+                    _channel_left  = animcurve_get_channel(_animation_curve, 0);
+                    _channel_right = animcurve_get_channel(_animation_curve, 0);
+                }
+                else
+                {
+                    _channel_left  = animcurve_get_channel(__haptic_curve, "left");
+                    _channel_right = animcurve_get_channel(__haptic_curve, "right");
```

Comment:
I'm not sure if forcing users to have `left` and `right` channel is good, they might have to use same curve for something else, that's why in my version I was checking channel 0 and 1 (that also required less setup from user side).

Reply:
I agree the user might want to use the curve for something else -- which is why I think using the first and second channels is too arbitrary :-)

Reply:
https://github.com/gnysek/Input/pull/3